### PR TITLE
REPOSITORY_STRUCTURE.md - blockstore description

### PR DESCRIPTION
**Pull request by qkrorlqr**

### CODE REVIEW

File: doc/REPOSITORY_STRUCTURE.md [hunk 1]
```diff
@@ -66,9 +66,73 @@
...
+[apps/disk_agent](/cloud/blockstore/apps/disk_agent) - storage node server
+[apps/endpoint_proxy](/cloud/blockstore/apps/endpoint_proxy) - a proxy which can maintain an NBD connection with the kernel and proxy the requests to nbsd (to allow nbsd updates without NBD device reconfiguration)
+[apps/server](/cloud/blockstore/apps/server) - the main NBS server (nbsd) - provides NBD/vhost-blk/gRPC APIs both for dataplane and controlplane ops, can host volume tablets
+[apps/server_lightweight](/cloud/blockstore/apps/server_lightweight) - nbsd version without YDB BlobStorage dependencies - uses local fs for storage, suitable for local debugging and as an nbsd mock to test your app's integration with nbsd
+[apps/tools](/cloud/blockstore/apps/tools) - misc tools for testing, debugging, log analysis and visualization
+
+### Blockstore libs
+[libs/client](/cloud/blockstore/libs/client) - client interface, gRPC client implementation, durable client (retrying client) implementation, session (remounting client) implementation, throttling client implementation
+[libs/client_rdma](/cloud/blockstore/libs/client_rdma) - client implementation over our RoCEv2-based transport lib
```
Comment (budevg) [inline doc/REPOSITORY_STRUCTURE.md:86]: actually there is nothing indicating we use RoCEv2 in the code so better just use `RDMA-based` here and in client

File: doc/REPOSITORY_STRUCTURE.md [hunk 2]
```diff
@@ -66,9 +66,73 @@
...
+[libs/endpoints_vhost](/cloud/blockstore/libs/endpoints_vhost) - vhost-blk endpoint listener
+[libs/kikimr](/cloud/blockstore/libs/kikimr) - YDB BlobStorage integration helpers
+[libs/kms](/cloud/blockstore/libs/kms) - KMS client API and implementation
+[libs/logbroker](/cloud/blockstore/libs/logbroker) - LogBroker (YDB BlobStorage-based persistent queue) client interface and topic API-based implementation
+[libs/nbd](/cloud/blockstore/libs/nbd) - NBD protocol, client and server
+[libs/notify](/cloud/blockstore/libs/notify) - user notification service integration
+[libs/nvme](/cloud/blockstore/libs/nvme) - NVMe specifics - getting device serial numbers, formatting devices, etc
+[libs/rdma](/cloud/blockstore/libs/rdma) - libibverbs and librdmacm-based transport lib, used for transport over RoCEv2
+[libs/rdma_test](/cloud/blockstore/libs/rdma_test) - mocks for testing the code that uses libs/rdma
+[libs/server](/cloud/blockstore/libs/server) - NBS gRPC server
```
Comment (budevg) [inline doc/REPOSITORY_STRUCTURE.md:109]: remove redundant space in the end of line

File: doc/REPOSITORY_STRUCTURE.md [hunk 3]
```diff
@@ -66,9 +66,73 @@
...
+
+### Blockstore libs/storage
 
 TODO
 
+### Blockstore tools
+[tools/analytics](/cloud/blockstore/tools/analytics) - various visualization, dumping and statistics calculation
+[tools/ci](/cloud/blockstore/tools/ci) - tools for e2e testing on real clusters
+[tools/cms](/cloud/blockstore/tools/cms) - tools for configuring NBS via YDB CMS
+[tools/csi_driver](/cloud/blockstore/tools/csi_driver) - [K8s CSI](https://kubernetes-csi.github.io/docs/) implementation for NBS & Filestore
```
Comment (budevg) [inline doc/REPOSITORY_STRUCTURE.md:129]: remove redundant space in the end of line

File: doc/REPOSITORY_STRUCTURE.md [hunk 4]
```diff
@@ -66,9 +66,73 @@
...
+[blockstore/tests](/cloud/blockstore/tests) - integration tests - most of them launch single-node BlobStorage cluster + NBS and then issue some requests and check the responses/metrics/etc
+
+### Blockstore apps
+[apps/client](/cloud/blockstore/apps/client) - cmdline client which uses mostly gRPC API - can perform almost any operation which is present in the API of nbsd
+[apps/disk_agent](/cloud/blockstore/apps/disk_agent) - storage node server
+[apps/endpoint_proxy](/cloud/blockstore/apps/endpoint_proxy) - a proxy which can maintain an NBD connection with the kernel and proxy the requests to nbsd (to allow nbsd updates without NBD device reconfiguration)
+[apps/server](/cloud/blockstore/apps/server) - the main NBS server (nbsd) - provides NBD/vhost-blk/gRPC APIs both for dataplane and controlplane ops, can host volume tablets
+[apps/server_lightweight](/cloud/blockstore/apps/server_lightweight) - nbsd version without YDB BlobStorage dependencies - uses local fs for storage, suitable for local debugging and as an nbsd mock to test your app's integration with nbsd
+[apps/tools](/cloud/blockstore/apps/tools) - misc tools for testing, debugging, log analysis and visualization
```
Comment (budevg) [inline doc/REPOSITORY_STRUCTURE.md:82]: missing `[blockstore/vhost-server](/cloud/blockstore/vhost-server) - vhost server providing access to local disks using aio lib or to remote disks using rdma lib`

**Review by budevg — CHANGES_REQUESTED** [review]

File: doc/REPOSITORY_STRUCTURE.md (hunk 4 above)
Comment (qkrorlqr) [reply doc/REPOSITORY_STRUCTURE.md:94]: it's not missing, but I accidentally put it in the wrong section (tools)
fixed

**Review by budevg — APPROVED** [review]

**Pull request merged**